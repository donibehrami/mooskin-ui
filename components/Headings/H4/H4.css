@import url('https://cdn.moostaging.com/assets/fonts/google-web-fonts/fonts.css');
@import '../../_utils/globals/variables.css';

.h4 {
    font-family: 'Convergence', 'Alegreya Sans', sans-serif;
    /*font-size: 15px;*/
    font-size: 0.9375rem;
    color: var(--smallHeading);
    margin: 19px 0;
}
