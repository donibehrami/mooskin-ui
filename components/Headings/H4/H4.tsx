import * as React from 'react';

import styles from './H4.css';

import {IHeadingProps} from '../props';

class H4 extends React.Component<IHeadingProps, void> {

    public static defaultProps = {
        className: '',
        style: {}
    };

    public render() {

        const {id, className, style, children} = this.props;

        const classes = `button-component ${styles.h4} ${className}`;

        return (
            <h4
                id={id}
                className={classes}
                style={style}
            >
                {children}
            </h4>
        );
    }
}

export default H4;
