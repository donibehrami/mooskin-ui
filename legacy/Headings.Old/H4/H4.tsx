import * as React from 'react';

import styles from './H4.css';

import {IHeadingProps} from '../props';

class H4 extends React.Component<IHeadingProps, {}> {

    static defaultProps = {
        className: '',
        style: {}
    };

    static displayName = 'H4';

    render() {

        const {id, className, style, children} = this.props;

        const classes = `h4-component ${styles.h4} ${className}`;

        return (
            <div
                id={id}
                className={classes}
                style={style}
            >
                {children}
            </div>
        );
    }
}

export default H4;
