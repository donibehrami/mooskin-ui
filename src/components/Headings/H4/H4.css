@import url('https://fonts.googleapis.com/css?family=Convergence');

.h4 {
    font-family: 'Convergence', sans-serif;
    font-weight: 400;
    color: #3a4257;
}