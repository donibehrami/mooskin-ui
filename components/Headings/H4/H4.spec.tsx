import * as React from 'react';
import H4 from './H4';

import { shallow } from 'enzyme';

describe('H4', () => {

    test('renders properly into dom with text, id and class', () => {

        const component = shallow(<H4 id="Campaigns" className="headers">Mooo!</H4>);

        expect(component.find('h4').text()).toBe('Mooo!');
        expect(component.find('h4').prop('id')).toEqual('Campaigns');
        expect(component.find('h4').hasClass('headers')).toBe(true);
    });

});
