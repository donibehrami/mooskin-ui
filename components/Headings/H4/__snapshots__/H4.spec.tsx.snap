// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H4 renders correctly 1`] = `
<h4
  className="h4-component h4 myClass"
  id="heading4"
  style={
    Object {
      "color": "blue",
    }
  }
>
  Mooskin
</h4>
`;
